### PR TITLE
fix:[CORE-1489] fix for subnet collector of azure data collector

### PR DESCRIPTION
**Pull request by raviteja-mandala**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

Fixes # (issue)

1. Fixed NPE by adding null check in subnet collector of azure.

2. Incase azure.alertRules().activityLogAlerts().list() throw null pointer exception, let us return empty list. We need not publish data alert in this case because it is the internal method azure apis.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

Tested locally.

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My commit message/PR follows the contribution guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java [hunk 1]
```diff
@@ -63,14 +63,23 @@ public List<SubnetVH> fetchSubnetDetails(SubscriptionVH subscription) {
...
+							subnetVH.setPrivateLinkServiceNetworkPolicies(
+									propertiesMap.get("privateLinkServiceNetworkPolicies").toString());
+						}
+						if(propertiesMap.get("provisioningState")!=null){
+							subnetVH.setProvisioningState(propertiesMap.get("provisioningState").toString());
+						}
+						if(propertiesMap.get("provisioningState")!=null){
+							subnetVH.setPrivateEndpointNetworkPolicies(
+									propertiesMap.get("privateEndpointNetworkPolicies").toString());
+						}
```
Comment (mkinchin-paladin) [inline jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java:82]: Pls reformat your code, basic InteliJ reformat is pretty much enough (Ctrl+Alt+L). If it's not available - use line 63 as an example.

Comment (mkinchin-paladin) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java:82]: Intead of `.get("asdfasdf") != null` use method `has` - its purpose is exactly what you need.

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Needs some minor tweaks

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java (hunk 1 above)
Comment (raviteja-mandala) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java]: Reformatted the code. 
Regarding second point, what if map has the key but has null as value.  map.containsKey("asdfasdf") return true but value may be still null, again we need an additional check. To be on safer side, lets explicitly check for null value.

**Review by santhosh-challa — COMMENTED** [review]
LGTM

**Review by santhosh-challa — APPROVED** [review]

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java (hunk 1 above)
Comment (mkinchin-paladin) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SubnetInventoryCollector.java]: oh, yup, missed it, you are right.

**Review by mkinchin-paladin — APPROVED** [review]
LGTM!

**Pull request merged**